src/main/java/seedu/address/model/history/VersionedAddressBook.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.history;
+
+import java.util.ArrayList;
+
+import seedu.address.model.AddressBook;
+
+public class VersionedAddressBook {
+    private ArrayList<AddressBook> addressBookStateList;
+    private int currentStatePointer;
+
+    public VersionedAddressBook() {
+        addressBookStateList = new ArrayList<>();
+        currentStatePointer = 0;
+    }
```

Comment:
i think there can be an initial state of empty or the loaded data

e.g. if you startup and run `add`, then you should be able to undo to the initial state at startup

then this way, your `currentStatePointer` can remain as 0, then this value actually refers to the index!